view/frontend/web/js/view/payment/method-renderer/vault.js

```diff
@@ -11,50 +11,244 @@
 /*global define*/
 define([
     'jquery',
-    'Magento_Vault/js/view/payment/method-renderer/vault'
-], function ($, VaultComponent) {
+    'ko',
+    'uiLayout',
+    'mage/url',
+    'Magento_Checkout/js/action/place-order',
+    'Magento_Checkout/js/model/full-screen-loader',
+    'Magento_Ui/js/model/messages',
+    'Magento_Vault/js/view/payment/method-renderer/vault',
+    'Adyen_Payment/js/model/adyen-payment-service',
+    'Adyen_Payment/js/model/adyen-configuration',
+    'Adyen_Payment/js/model/adyen-checkout',
+    'Adyen_Payment/js/model/adyen-payment-modal'
+], function (
+    $,
+    ko,
+    layout,
+    url,
+    placeOrderAction,
+    fullScreenLoader,
+    Messages,
+    VaultComponent,
+    adyenPaymentService,
+    adyenConfiguration,
+    adyenCheckout,
+    adyenPaymentModal
+) {
     'use strict';
 
+    let isValidObserver = ko.observable(false);
+    let validTokens = {};
+
     return VaultComponent.extend({
         defaults: {
-            template: 'Magento_Vault/payment/form'
+            template: 'Adyen_Payment/payment/card-vault-form.html',
+            checkoutComponentBuilt: false,
+            modalLabel: 'card_action_modal'
+        },
+
+        initObservable: function () {
+            this._super()
+                .observe([
+                    'checkoutComponentBuilt'
+                ]);
+
+            return this;
+        },
+
+        initialize: function () {
+            let self = this;
+            this._super();
+            let paymentMethodsObserver = adyenPaymentService.getPaymentMethods();
+            paymentMethodsObserver.subscribe(
+                function (paymentMethodsResponse) {
+                    self.loadCheckoutComponent(paymentMethodsResponse)
+                });
+            self.loadCheckoutComponent(paymentMethodsObserver());
+
+            return this;
+        },
+
+        loadCheckoutComponent: async function(paymentMethodsResponse) {
+            this.checkoutComponent = await adyenCheckout.buildCheckoutComponent(
+                paymentMethodsResponse,
+                this.handleOnAdditionalDetails.bind(this)
+            );
+
+            if (this.checkoutComponent) {
+                this.checkoutComponentBuilt(true)
+            }
+        },
+
+        // TODO: implement handleOnAdditionalDetails
```

Comment:
Since `handleOnAdditionalDetails` wasn't implemented, what do you think about merging it to a feature branch first?

Reply:
I don't think we need to cause this won't interfere with any other features/tasks that are not vault-related. We should add this in an upcoming PR.